src/botpress.js

```diff
@@ -195,7 +195,7 @@ class botpress {
     const dialogStateManager = StateManager({ db })
     const flowProvider = new FlowProvider({ logger, projectLocation, botfile, ghostManager })
     const dialogJanitor = new DialogJanitor({ db, middlewares, botfile })
-    const dialogEngine = new DialogEngine(flowProvider, dialogStateManager, null, logger)
+    const dialogEngine = new DialogEngine(flowProvider, dialogStateManager, notifications, null, logger)
```

Comment:
Let's make it accept "named" arguments with object / desctructuring? It already has too many of them.
Also if we don't use `oiptions` maybe let's drop it? Or is the engine also instantiated in some other place?

Reply:
Hm, looks like options are not even used in the constructor there

Reply:
@emirotin , OK refactored it to be named params.
As for options - not sure. @slvnperron , are we planning to use them?